### PR TITLE
Updated Routes

### PR DESCRIPTION
**Pull request by Yeshuwa**
Updated route assignment!

### CODE REVIEW

File: src/Events/Events.ts [hunk 1]
```diff
@@ -3,6 +3,145 @@ import Requests from "../Util/Requests";
...
+    };
+    private static routeConnectUser: Route = { 
+        route : "/events/{id}/connectUser",
+        method : RequestTypes.POST
+    };
+    private static routeAddUserToAudience: Route = { 
+        route : "/events/{id}/removeUserFromStage",
+        method : RequestTypes.POST
+    };
+    private static
```
Comment (ProdigyView) [inline src/Events/Events.ts]: This is a bug right here. The code will not compile, please fix.

File: src/Events/Events.ts [hunk 2]
```diff
@@ -312,6 +451,144 @@ class Events {
         return Requests.post(route, data, query, options);
     }
 
+    public static createGroup (event_id : string, data : object,query? : object | null,options? : object |null  ) {
+        let route = this.routeCreateGroup.route.replaceAll('{id}',event_id);
+        
+        return Requests.post(route,data,query,options);
+    } 
+    public static updateGroup(event_id : string, data : object, query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: There should be another parameter, group_id. And the parameter should be replaced in the route as well.

File: src/Events/Events.ts [hunk 3]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        let route = this.routeUpdateGroup.route.replaceAll('{id}', event_id);
+        return Requests.put(route, data, query, options);
+    }
+    public static listGroup(event_id : string,  query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: Again, the group_id should be a parameter, and also replaced in the route.

File: src/Events/Events.ts [hunk 4]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        let route = this.routeListGroup.route.replaceAll('{id}', event_id);
+
+        return Requests.get(route, query, options);
+    }
+    public static ViewGroup(event_id : string,  query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: Same here the group_id should be a parameter, and replaced in the route.

File: src/Events/Events.ts [hunk 5]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        let route = this.routeViewGroup.route.replaceAll('{id}', event_id);
+
+        return Requests.get(route, query, options);
+    }
+    public static deleteGroup(data : object, query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: You are forgetting to pass the event_id and group id, and replacing those both in the route.

File: src/Events/Events.ts [hunk 6]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        return Requests.post(this.routeDeleteGroup.route, data, query, options);
+    }
+    public static registerAttendee(data : object, query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: The event id in the route is missing

File: src/Events/Events.ts [hunk 7]
```diff
@@ -3,6 +3,145 @@ import Requests from "../Util/Requests";
...
+    private static routeRefundTicket: Route = {
+        route : "/events/{event_id}/tickets/{ticket_id}/refund",
+        method : RequestTypes.POST
+    }
+    private static routeViewTicket: Route = {
+        route : "/events/{event_id}/tickets/{ticket_id}",
+        method : RequestTypes.GET
+    }
+    private static routeHasTicket: Route = {
+        route : "/events/{event_id}/hasTicket/{account_id",
```
Comment (ProdigyView) [inline src/Events/Events.ts]: account id is missing ending bracket

File: src/Events/Events.ts [hunk 8]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+    }
+    public static registerAttendee(data : object, query? : object | null, options? : object | null) {
+        return Requests.post(this.routeRegisterAttendee.route, data, query, options);
+    }
+    public static listTickets(event_id : string,  query? : object | null, options? : object | null) {
+        let route = this.routeListTickets.route.replaceAll('{id}', event_id);
+
+        return Requests.get(route, query, options);
+    }
+    public static hasTicket(event_id : string,  query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: The account id is required in the parameter, and has to be replaced in the route.

File: src/Events/Events.ts [hunk 9]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        let route = this.routeHasTicket.route.replaceAll('{id}', event_id);
+
+        return Requests.get(route, query, options);
+    }
+    public static viewTicket(event_id : string,  query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: The ticket_id is required as the parameter, and must be replaced in the route.

File: src/Events/Events.ts [hunk 10]
```diff
@@ -312,6 +451,144 @@ class Events {
...
+        let route = this.routeViewTicket.route.replaceAll('{id}', event_id);
+
+        return Requests.get(route, query, options);
+    }
+    public static refundTicket(data : object, query? : object | null, options? : object | null) {
```
Comment (ProdigyView) [inline src/Events/Events.ts]: The event_id and ticket_id is missing the parameters, and must be replaced.

**Review by ProdigyView — CHANGES_REQUESTED** [review]
So I didn't go through the entire thing but a lot of the bugs in here is that you are forgetting both parameters and to replace them in the routes. For example:

`/events/{event_id}/tickets/{ticket_id}/refund",

This route has two variables that need replacing, event_id and ticket_id . So your method should look like the following:

```
public static refundTicket(event_id : string, ticket_id : string, data : object, query? : object | null, options? : object | null) {
    let route = this.routeCreateGroup.route.replaceAll('{id}',event_id);
    route = route.replaceAll('{ticket_id}',ticket_id);

}
```

Before sure to review the code and check for all of those carefully.

File: src/Events/Events.ts (hunk 1 above)
Comment (ProdigyView) [reply src/Events/Events.ts]: @Yeshuwa This bug has still not been addressed.

File: src/Events/Events.ts (hunk 7 above)
Comment (ProdigyView) [reply src/Events/Events.ts]: @Yeshuwa Still need to address this issue.

File: src/Events/Events.ts (hunk 2 above)
Comment (ProdigyView) [reply src/Events/Events.ts]: How is this resolve? group_id is still missing.

File: src/Events/Events.ts (hunk 9 above)
Comment (ProdigyView) [reply src/Events/Events.ts]: Same here

**Review by ProdigyView — CHANGES_REQUESTED** [review]
The changes that we talked about have not been done. What happened?

**Comment by Yeshuwa** [comment]
Maybe I missed some of them, but all the ones that you assigned through
GitHub and left a message for me for I fixed. I’ll go back and finish them
today along with the react assignment.

**Pull request merged**